### DISCUSSION TITLE
Strategist for running C, C++ test cases from one runner?

### DISCUSSION THREAD

**Discussion by dreamer-coding** [mesonbuild/meson#13849] [Q&A] [answered]
Are there strategist for running both C and C++ test cases from one test runner? Lets assume a given project is written in C and happens to be compatible with C++, skipping all of the detail of how this part is done to save time, how can this project be tested with test cases for two languages and be ran with one test runner? If this is impossible I'm open to a more logical solution for running test cases for both C and C++.

**Comment by eli-schwartz** [comment, marked as answer]
If meson is your test runner then it would simply execute the C test as one executable, and the C++ test as a different executable, and expect both test programs to return success.

You could test a C library by writing C interface tests in one program, and writing C++ interface tests in a second program, to test that the library still works when compiled and used as C++.

**Reply by dreamer-coding** [reply]
Alright, thanks.

**Reply by dreamer-coding** [reply]
I did it anyway, just had to do a few small changes to run test cases under a single runner though still appreciate the answer.
